### PR TITLE
fix: nullish examples in examples.json

### LINKED ISSUE #994: Having invalid examples in spec fails CLI build
#### Describe the bug

The script that CLI runs in its build process fetches examples from the spec repository, If any example in the spec is invalid the `examples.json` file that is being created by the script creates a null value in the script and thus the build fails. Even older versions of the CLI start having issues due to this. 

#### How to Reproduce
The only way to reproduce is if we have an invalid example in the spec repository or by copy/pasting this `example.json` file. 

```json
[
    {
        "name": "Account Service",
        "value": "simple-asyncapi.yml"
    },
    {
        "name": "Account Service",
        "value": "simple.yml"
    },
    {
        "name": "AnyOf example",
        "value": "anyof-asyncapi.yml"
    },
    {
        "name": "AnyOf example",
        "value": "anyof.yml"
    },
    {
        "name": "Application Headers example - (protocols: mqtt)",
        "value": "application-headers-asyncapi.yml"
    },
    {
        "name": "Application Headers example - (protocols: mqtt)",
        "value": "application-headers.yml"
    },
    {
        "name": "Correlation ID Example - (protocols: mqtt)",
        "value": "correlation-id-asyncapi.yml"
    },
    {
        "name": "Correlation ID Example - (protocols: mqtt)",
        "value": "correlation-id.yml"
    },
    {
        "name": "Gemini Market Data Websocket API - (protocols: wss)",
        "value": "websocket-gemini-asyncapi.yml"
    },
    {
        "name": "Gemini Market Data Websocket API - (protocols: wss)",
        "value": "websocket-gemini.yml"
    },
    {
        "name": "Gitter Streaming API - (protocols: https)",
        "value": "gitter-streaming-asyncapi.yml"
    },
    {
        "name": "Gitter Streaming API - (protocols: https)",
        "value": "gitter-streaming.yml"
    },
    {
        "name": "Kraken Websockets API",
        "value": "kraken-websocket-request-reply-message-filter-in-reply-asyncapi.yml"
    },
    {
        "name": "Kraken Websockets API",
        "value": "kraken-websocket-request-reply-multiple-channels-asyncapi.yml"
    },
    {
        "name": "Mercure Hub Example - (protocols: mercure)",
        "value": "mercure-asyncapi.yml"
    },
    {
        "name": "Mercure Hub Example - (protocols: mercure)",
        "value": "mercure.yml"
    },
    {
        "name": "Not example",
        "value": "not-asyncapi.yml"
    },
    {
        "name": "Not example",
        "value": "not.yml"
    },
    {
        "name": "Notifications",
        "value": "operation-security-asyncapi.yml"
    },
    {
        "name": "Notifications",
        "value": "operation-security.yml"
    },
    {
        "name": "OneOf example",
        "value": "oneof-asyncapi.yml"
    },
    {
        "name": "OneOf example",
        "value": "oneof.yml"
    },
    {
        "name": "RPC Client Example - (protocols: amqp)",
        "value": "rpc-client-asyncapi.yml"
    },
    {
        "name": "RPC Client Example - (protocols: amqp)",
        "value": "rpc-client.yml"
    },
    {
        "name": "RPC Server Example - (protocols: amqp)",
        "value": "rpc-server-asyncapi.yml"
    },
    {
        "name": "RPC Server Example - (protocols: amqp)",
        "value": "rpc-server.yml"
    },
    {
        "name": "Slack Real Time Messaging API - (protocols: https)",
        "value": "slack-rtm-asyncapi.yml"
    },
    {
        "name": "Slack Real Time Messaging API - (protocols: https)",
        "value": "slack-rtm.yml"
    },
    {
        "name": "Streetlights App - (protocols: mqtt)",
        "value": "tutorial.yml"
    },
    {
        "name": "Streetlights Kafka API - (protocols: kafka-secure,kafka-secure)",
        "value": "streetlights-kafka-asyncapi.yml"
    },
    {
        "name": "Streetlights Kafka API - (protocols: kafka-secure,kafka-secure)",
        "value": "streetlights-kafka.yml"
    },
    {
        "name": "Streetlights Kafka API - (protocols: kafka-secure,kafka-secure)",
        "value": "streetlights-operation-security-asyncapi.yml"
    },
    {
        "name": "Streetlights Kafka API - (protocols: kafka-secure,kafka-secure)",
        "value": "streetlights-operation-security.yml"
    },
    {
        "name": "Streetlights MQTT API - (protocols: mqtt)",
        "value": "streetlights-mqtt-asyncapi.yml"
    },
    {
        "name": "Streetlights MQTT API - (protocols: mqtt)",
        "value": "streetlights-mqtt.yml"
    },
    null
]

```

#### Expected behavior
If we have any invalid example we should not generate the `example.json` with a null value.

### PR DESCRIPTION
**Pull request by Shiva953**
**Description**

- This PR addresses the issue related to the async-api CLI build process failing when encountering invalid examples in the spec repository. It filters out the null values from the exampleList.
- ...

**Related issue(s)**
Resolves #994

### CODE REVIEW

**Comment by Shiva953** [comment]
@smoya

File: scripts/fetch-asyncapi-example.js [hunk 1]
```diff
@@ -86,7 +86,7 @@ const buildCLIListFromExamples = async () => {
     }
   });
 
-  const exampleList = await Promise.all(buildExampleList);
+  const exampleList = (await Promise.all(buildExampleList)).filter(item => item!==null);
```
Comment (smoya) [inline scripts/fetch-asyncapi-example.js:89]: Nitpick: 
```suggestion
  const exampleList = (await Promise.all(buildExampleList)).filter(item => item !== null);
```

It seems we are not passing any linter to the scripts under `scripts` dir (I don't know the reason cc @Souvikns). Anyway this nitpick I suggest improves readability and needs to be considered.

**Comment by smoya** [comment]
@Shiva953 could you confirm you have tested this with an `examples.json` file that includes a `null` object?

**Comment by Shiva953** [comment]
> @Shiva953 could you confirm you have tested this with an `examples.json` file that includes a `null` object?

sure, here's the test(however, I also had to made the condition `if(!document)` to `return null;` instead of just `return;`, because seemed like it was returning undefined instead of null before, which led the test to fail before it. Tho now after the changes it works fine.


https://github.com/asyncapi/cli/assets/120790871/2627ad9c-71c6-426e-a099-7ad19f44cfdc

File: scripts/fetch-asyncapi-example.js [hunk 2]
```diff
@@ -82,11 +82,11 @@ const buildCLIListFromExamples = async () => {
       };
     } catch (error) {
       // Failed for somereason to parse this spec file, ignore for now
-      console.error(error);
```
Comment (smoya) [inline scripts/fetch-asyncapi-example.js:85]: I would keep this. No reason to remove it IMHO as we want to know if something goes wrong anyway.

File: scripts/fetch-asyncapi-example.js [hunk 3]
```diff
@@ -82,11 +82,11 @@ const buildCLIListFromExamples = async () => {
...
+      return null;
     }
   });
 
-  const exampleList = await Promise.all(buildExampleList);
+  const exampleList = (await Promise.all(buildExampleList)).filter(item => item !== null);
```
Comment (smoya) [inline scripts/fetch-asyncapi-example.js:89]: Using double negation would make this compatible with any scenario. Then you won't need to change the return [here](https://github.com/asyncapi/cli/pull/1033/files#diff-3796d0c36710ab426831cd95695d7165c7fe53517159316d107a3678bdb4705fR74) to be explicitly a `null` 
```suggestion
  const exampleList = (await Promise.all(buildExampleList)).filter(item => !!item);
```

Does it make sense? WDYT?

File: scripts/fetch-asyncapi-example.js (hunk 2 above)
Comment (Shiva953) [reply scripts/fetch-asyncapi-example.js:85]: makes sense

**Comment by smoya** [comment]
We need a code owner to aprove the workflows to run, also to review this. cc @Souvikns @magicmatatjahu @derberg

**Review by Souvikns — APPROVED** [review]

**Comment by Shiva953** [comment]
@smoya any changes that need to be made?

**Review by Souvikns — APPROVED** [review]

**Pull request merged**